jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/common/BaseCharacteristic.kt

```diff
@@ -11,31 +11,35 @@ import kotlin.jvm.java
  *
  * Created by miantorno on 2018-06-19.
  */
-abstract class BaseCharacteristic(characteristic: BluetoothGattCharacteristic?, val uuid: Int) {
-    open val tag = BaseCharacteristic::class.java.canonicalName as String
-    val nullValueException = "Null characteristic interpretation, aborting parsing."
-    val rawData: ByteArray = characteristic?.value ?: ByteArray(0)
-    var successfulParsing: Boolean = false
-    var offset = 0
+abstract class BaseCharacteristic(val uuid: Int) {
 
     /**
      * If the characteristic is null, which is possible, we just return false so the requesting
      * process can handle the failed parse. This is done in the super class, so we don't have to
      * deal with it in every other sub class characteristic.
      */
-    init {
+    constructor(characteristic: BluetoothGattCharacteristic?, uuid: Int): this(uuid) {
+        this.characteristic = characteristic
         characteristic?.let {
+            rawData = it.value ?: ByteArray(0)
             this.successfulParsing = tryParse(it)
         }
     }
 
+    open val tag = BaseCharacteristic::class.java.canonicalName as String
+    val nullValueException = "Null characteristic interpretation, aborting parsing."
+    var rawData: ByteArray = ByteArray(0)
+    var characteristic: BluetoothGattCharacteristic? = null
+    var successfulParsing: Boolean = false
+    var offset = 0
+
     /**
      * Swallowing the exception is one of two viable options, some users might want any error to
      * bubble up all the way immediately.
      *
      * https://github.com/markiantorno/JDRFAndroidBLEParser/issues/1
      */
-    protected fun tryParse(c: BluetoothGattCharacteristic): Boolean {
+    fun tryParse(c: BluetoothGattCharacteristic): Boolean {
```

Comment:
We want it to be public now so that the characteristic can be hand loaded.
 
It used to be done through the constructor, but now it's done either through the constructor or through the method explicitly.

Why do you think this would be a security issue? Is there something I should be taking into account?

Reply:
Now I think of it. I guess your intention is to allow a compose() method be called explicitly, so you make it the same for tryParse(), am I right?

Reply:
That is correct